token/js/src/instructions/uiAmountToAmount.ts

```diff
@@ -0,0 +1,123 @@
+import { u8, struct, cstr} from '@solana/buffer-layout';
+import { AccountMeta, PublicKey, TransactionInstruction } from '@solana/web3.js';
+import { TOKEN_PROGRAM_ID } from '../constants';
+import {
+    TokenInvalidInstructionDataError,
+    TokenInvalidInstructionKeysError,
+    TokenInvalidInstructionProgramError,
+    TokenInvalidInstructionTypeError,
+} from '../errors';
+import { TokenInstruction } from './types';
+
+/** TODO: docs */
+export interface UiAmountToAmountInstructionData {
+    instruction: TokenInstruction.UiAmountToAmount;
+    amount: string;
+}
+
+/** TODO: docs */
+export const uiAmountToAmountInstructionData = struct<UiAmountToAmountInstructionData>([u8('instruction'),cstr('amount')]);
+
+/**
+ * Construct a UiAmountToAmount instruction
+ *
+ * @param mint         Public key of the mint
+ * @param amount       UiAmount of tokens to be converted to Amount
+ * @param programId    SPL Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createUiAmountToAmountInstruction(
+    mint: PublicKey,
+    amount: string,
+    programId = TOKEN_PROGRAM_ID
+): TransactionInstruction {
+    const keys = [{ pubkey: mint, isSigner: false, isWritable: true }];
```

Comment:
The mint isn't being written to here, so be sure to keep it read-only to help with parallelization of transactions on the network
```suggestion
    const keys = [{ pubkey: mint, isSigner: false, isWritable: false }];
```